src/main/java/step1/SignKind.java

```diff
@@ -0,0 +1,33 @@
+package step1;
+
+import java.util.Arrays;
+import java.util.function.BiFunction;
+
+
+public enum SignKind {
+
+    PLUS("+", (x, y) -> x + y),
+    MINUS("-", (x, y) -> x - y),
+    MULTIPLY("*", (x, y) -> x * y),
+    DIVIDE("/", (x, y) -> x / y);
+
+    private BiFunction<Integer, Integer, Integer> expression;
+
+    //기호
+    private String sign;
+
+    SignKind(String sign, BiFunction<Integer, Integer, Integer> expression) {
+        this.sign = sign;
+        this.expression = expression;
+    }
+
+    public Integer calculate(Integer x, Integer y) {
+        return getSign(sign).expression.apply(x,y);
+    }
+
+    public static SignKind getSign(String sign){
+        return Arrays.stream(values()).filter(o->o.sign.equals(sign)).findFirst().orElseThrow(()-> new IllegalArgumentException("올바른 문자열이 아닙니다."));
```

Comment:
개인적으로 메서드 체이닝 사용시 가독성을 위해 한줄에 하나의 dot(.)만 사용하는게 좋다고 생각해요.
```suggestion
        return Arrays.stream(values())
          .filter(o->o.sign.equals(sign))
          .findFirst()
          .orElseThrow(()-> new IllegalArgumentException("올바른 문자열이 아닙니다."));
```

Reply:
그렇네요! 이렇게 하니깐 가독성이 좋네요! ㅋㅋㅋ